python/lsst/daf/butler/registry/queries/expressions/check.py

```diff
@@ -194,10 +194,11 @@ def visitIdentifier(self, name: str, node: Node) -> TreeSummary:
         # Docstring inherited from TreeVisitor.visitIdentifier
         if name in self.bindKeys:
             return TreeSummary()
-        if categorizeIngestDateId(name):
-            return TreeSummary(
-                hasIngestDate=True,
-            )
+        constant = categorizeConstant(name)
+        if constant is ExpressionConstant.INGEST_DATE:
+            return TreeSummary(hasIngestDate=True)
+        elif constant is not None:
+            return TreeSummary()
```

Comment:
Maybe try to make it a bit more stable w.r.t. potential extension of ExpressionConstant, like:

    if constant is ExpressionConstant.INGEST_DATE:
        ...
    elif constant is ExpressionConstant.NULL:
        ....
    assert constant is None, "..."